src/blockdata/transaction.rs

```diff
@@ -648,6 +668,38 @@ impl Transaction {
     pub fn is_explicitly_rbf(&self) -> bool {
         self.input.iter().any(|input| input.sequence < (0xffffffff - 1))
     }
+
+    /// Returns true if this [`Transaction`] is final i.e., can be included in the blockchain at
+    /// `height`/`time`.
+    ///
+    /// This is `isFinalTx` from [Bitcoin Core].
+    ///
+    /// [Bitcoin Core]: https://github.com/bitcoin/bitcoin/blob/b9122e95f0f4ff5d2b2e21a5caf6c69d488c0347/src/consensus/tx_verify.cpp#L17
+    pub fn is_final(&self, height: Height, time: Time) -> bool {
+        if self.lock_time == locktime::ZERO {
+            return true;
+        }
```

Comment:
In be119a025a8930ad60985a2c2da0975329b945ba:

I think we should remove this check, it's redundant with the check below and potentially confusing.